### PR TITLE
✨ Sorting and Filtering Menu (#126)

### PR DESCRIPTION
**Pull request by Aymerix01**
The user can filter and sort all his files in every window with the file browsing. There are filters for extension, creation date and modification date. It is possible to sort the file and folder by alphabetic order, anti-alphabetic order, ascending and descending creation date and modification date.

This is how the UI looks like:
MenuTri

In case the screen is too small, the menu changes like that:
MenuTriSmall

And filters can be changed with a new modal:
ModalFilterMenu

### CODE REVIEW

**Comment by Aymerix01** [comment]
How it looks like:

https://github.com/linagora/TDrive/assets/123820772/edded17d-3f6f-420b-9ec9-7680d70fe60d

File: tdrive/frontend/src/app/features/router/services/router-service.ts [hunk 1]
```diff
@@ -50,7 +51,10 @@ class RouterServices extends Observable {
   clientSubPathnames: Readonly<string[]> = [
     '/client/:companyId',
     '/client/:companyId/v/:viewId',
+    '/client/:companyId/v/:viewId/filter/:filterSortId',
```
Comment (shepilov) [inline tdrive/frontend/src/app/features/router/services/router-service.ts]: make filter not a path variable, but a query param, 
so it should be ?filter=...

File: tdrive/frontend/src/app/features/users/services/filter-service.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
+import { DriveItemDetails} from "features/drive/types";
+import {Filter} from "features/drive/state/filter";
+import Languages from "features/global/services/languages-service";
+import useRouteState from "features/router/hooks/use-route-state";
+
```
Comment (shepilov) [inline tdrive/frontend/src/app/features/users/services/filter-service.tsx]: It should be .ts file and it should be a class

File: tdrive/frontend/src/app/features/users/services/filter-service.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    } else if (date === 'Last month'){
+        return oneMonthStamp
+    } else {
+        return 0
+    }
+}
+
+export function setFilterWithURL(filterSortId: string){
+    const filters = filterSortId!.split('&')
+    const sol = [
```
Comment (shepilov) [inline tdrive/frontend/src/app/features/users/services/filter-service.tsx]: what does it mean sol?

File: tdrive/frontend/src/app/features/users/services/filter-service.tsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    [
+        {key: Languages.t('components.item_context_menu.default'), value: 'All'},
+        {key: Languages.t('components.item_context_menu.alphabetical_order'), value: 'alphabetical_order'},
+        {key: Languages.t('components.item_context_menu.anti_alphabetical_order'), value: 'anti_alphabetical_order'},
+        {key: Languages.t('components.item_context_menu.ascending_modification_date'), value: 'ascending_modification_date'},
+        {key: Languages.t('components.item_context_menu.descending_modification_date'), value: 'descending_modification_date'},
+    ]]
+
+    filterInfo.forEach(function (list, i) {
+        list.forEach((dic) => {
```
Comment (shepilov) [inline tdrive/frontend/src/app/features/users/services/filter-service.tsx]: use data structure when you can get a current filter just by identifier, without double "for"

**Comment by Aymerix01** [comment]
https://github.com/linagora/TDrive/assets/123820772/afccfba0-bd37-4147-879d-61953c93fd6e

**Pull request closed without merging**